share/smack/top.py

```diff
@@ -891,16 +904,82 @@ def verification_result(verifier_output, verifier):
         return VResult.UNKNOWN
 
 
+def get_result(result):
+    return result
+
+
 def verify_bpl(args):
     """Verify the Boogie source file with a back-end verifier."""
 
+    # ugly way of adding additional params to the verifier options
+    if isinstance(args, tuple):
+        args, commands_to_add = args
+        args.verifier = commands_to_add['verifier']
+        args.verifier_options = commands_to_add['verifier-options']
+
+    if args.verifier == 'portfolio':
+        if args.portfolio_config is None:
+            yaml_info = {'thread1':
+                            {'verifier': 'corral',
+                             'verifier-options': ""},
+                         'thread2':
+                            {'verifier': 'corral',
+                             'verifier-options':
+                                 "/bopt:proverOpt:O:smt.arith.solver=2"},
+                         'thread3':
+                            {'verifier': 'corral',
+                             'verifier-options':
+                                 "/bopt:proverOpt:O:smt.qi.eager_threshold=100"},
+                         'thread4':
+                            {'verifier': 'corral',
+                             'verifier-options':
+                                 ("/bopt:proverOpt:O:smt.qi.eager_threshold=100"
+                                  " /bopt:proverOpt:O:smt.arith.solver=2")}}
+        else:
+            yaml_info = yaml.load(open(args.portfolio_config, 'r'))
+        p = multiprocessing.Pool()
+        thread_args = copy.deepcopy(args)
+
+        results = {}  # map of result -> thread name
```

Comment:
The item that I am checking on to see if "is done" is the AsyncResult object (result). Once I get the finished result I can then lookup its name by using the object itself as the key. I could flip this if that is more intuitive, it will just mean I will need to do a lookup in the dictionary every time I want to check "is done?".